qiskit/visualization/counts_visualization.py

```diff
@@ -268,9 +268,24 @@ def _plot_histogram_data(data, labels, number_to_keep):
 
     all_pvalues = []
     all_inds = []
+    # if multiple executions, we consider number_to_keep for each execution
+    # and this may result in more than number_to_keep slots
+    multiple_exec_keys_dict = OrderedDict()
```

Comment:
You're using an OrderedDict here. I'm not quite sure if it's relevant, but have you checked that this works if the keys are in different orders in different dictionaries?

Reply:
@drewrisinger you're right. No need for OrderedDict here.

Reply:
OrderedDict is essentially the default in ``python 3.7+`` anyways.